main.go

```diff
@@ -0,0 +1,155 @@
+	// to ensure that exec-entrypoint and run can make use of them.
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	oconfigv1 "github.com/openshift/api/config/v1"
+	monitorv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+
+	// OSD metrics
+	osdmetrics "github.com/openshift/operator-custom-metrics/pkg/metrics"
+
+	ocmagentmanagedopenshiftiov1alpha1 "github.com/openshift/ocm-agent-operator/api/v1alpha1"
+	"github.com/openshift/ocm-agent-operator/controllers/ocmagent"
+	//+kubebuilder:scaffold:imports
+)
+
+var (
+	scheme   = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup")
+)
+
+var (
+	osdMetricsPort = "8686"
+	osdMetricsPath = "/metrics"
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+
+	utilruntime.Must(ocmagentmanagedopenshiftiov1alpha1.AddToScheme(scheme))
+	utilruntime.Must(oconfigv1.Install(scheme))
+	utilruntime.Must(monitorv1.AddToScheme(scheme))
+	//+kubebuilder:scaffold:scheme
+}
+
+func main() {
+	var metricsAddr string
+	var enableLeaderElection bool
+	var probeAddr string
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
+	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
+	opts := zap.Options{
+		Development: true,
+	}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
+
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+
+	operatorNS, err := namespace.GetOperatorNamespace()
+	if err != nil {
+		setupLog.Error(err, "unable to determine operator namespace, please define OPERATOR_NAMESPACE")
+		os.Exit(1)
+	}
+
+	if err := monitorv1.AddToScheme(clientgoscheme.Scheme); err != nil {
+		setupLog.Error(err, "unable to add monitoringv1 scheme")
+		os.Exit(1)
+	}
+
+	metricsServer := osdmetrics.NewBuilder(operatorNS, "ocm-agent-operator").
+		WithPort(osdMetricsPort).
+		WithPath(osdMetricsPath).
+		WithCollectors(localmetrics.MetricsList).
+		WithServiceMonitor().
+		GetConfig()
+
+	if err := osdmetrics.ConfigureMetrics(context.TODO(), *metricsServer); err != nil {
+		setupLog.Error(err, "Failed to configure OSD metrics")
+	}
+
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Namespace:              operatorNS,
+		Scheme:                 scheme,
+		MetricsBindAddress:     metricsAddr,
+		Port:                   9443,
+		HealthProbeBindAddress: probeAddr,
+		LeaderElection:         enableLeaderElection,
+		LeaderElectionID:       "8716512f.managed.openshift.io",
```

Comment:
what is the ID here?

Reply:
@bmeng  This is generated by kubebuilder.  And this is something must be provided as per [controller-runtime ref](https://github.com/kubernetes-sigs/controller-runtime/blob/release-0.11/pkg/leaderelection/leader_election.go#L67-L70)
Ref: 
https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/plugins/golang/v2/scaffolds/internal/templates/main.go#L227